### PR TITLE
Add support for `-J` option group

### PR DESCRIPTION
**Pull request by saulecabrera**
This commit adds support for the `-J` option group and enables `javy-json` and `override-json-parse-and-stringify` by default.

Additionally, this change improves how the javy cli integration tests are defined by introducing a `javy_cli_test` which:

* Produces a test per supported command (`compile` and `build`)
* Provides a set of defaults depending on the command under test

Generating a test per command improves the developer experience of determining failures on particular command configurations.

Lastly, this change bumps the provider to v3, given that new funcionality is introduced as part of enabling `javy-json` and `override-json-parse-and-stringify`. In order to test that the `compile` command remains "frozen", using version 2 of the provider, this commit introduces a static copy of the v2 provided downloaded from the releases section.

- [x] I've updated the relevant CHANGELOG files if necessary. Changes to `javy-cli` and `javy-core` do not require updating CHANGELOG files.
- [x] I've updated the relevant crate versions if necessary. [Versioning policy for library crates](https://github.com/bytecodealliance/javy/blob/main/docs/contributing.md#versioning-for-library-crates)
- [x] I've updated documentation including crate documentation if necessary.

### CODE REVIEW

**Comment by saulecabrera** [comment]
The failures in CI are caused by https://github.com/rust-lang/rust/issues/123748; in CI, we're using the latest stable, which got updated today. I can repro those failures locally by updating the toolchain.

File: crates/cli/src/commands.rs [hunk 1]
```diff
@@ -228,6 +237,17 @@ option_group! {
...
         /// Whether to redirect the output of console.log to standard error.
         RedirectStdoutToStderr(bool),
+        /// Whether to enable the `Javy.JSON` builtins.
+        JavyJson(bool),
+        /// Whether to enable the `Javy.readSync` and `Javy.writeSync` builtins.
+        JavyStreamIo(bool),
+        /// Whether to override the `JSON.parse` and `JSON.stringify`
+        /// implementations with an alternative, more performant, SIMD based
+        /// implemetation.
+        OverrideJsonParseAndStringify(bool),
```
Comment (surma) [inline crates/cli/src/commands.rs:247]: Should this rather be called `SimdJsonBuiltins` or something along those lines?

File: crates/runner/src/lib.rs [hunk 2]
```diff
@@ -52,6 +62,11 @@ impl Default for Builder {
             preload: None,
             command: JavyCommand::Build,
             redirect_stdout_to_stderr: None,
+            javy_stream_io: None,
```
Comment (surma) [inline crates/runner/src/lib.rs:65]: Not really related to this PR, but for future simplicity, but might be worth it to have the `Builder` autogenerated using [`derive-builder`](https://docs.rs/derive_builder/0.20.1/derive_builder/)

File: crates/cli/src/commands.rs (hunk 1 above)
Comment (saulecabrera) [reply crates/cli/src/commands.rs]: Great suggestion, done!

**Pull request merged**